docs/details/main-components/draw.rst

```diff
@@ -4,77 +4,194 @@
 Drawing Pipeline
 ================
 
-LVGL has a flexible and extendable drawing pipeline.  You can hook it to do
-some rendering with a GPU or even completely replace the built-in
-software renderer.
-
 
 Overview
 ********
 
-Draw task
----------
-
-
-When :cpp:expr:`lv_draw_rect`, :cpp:expr:`lv_draw_label` or similar functions are called
-LVGL creates a so called draw task.
+Drawing is writing pixel colors into a buffer where they will be delivered to a
+display panel as pixels.  Sometimes it involves computing those colors before they
+are written (e.g. combining them with other colors when an object has partial opacity).
+
+On modern computing hardware meant to be used with larger display panels, there are
+sometimes options for different ways drawing can be accomplished.  For example, some
+MCUs come with hardware that is very good (and fast) at certain types of drawing
+tasks.  Alternatively, you might have access to a drawing library that performs
+certain types of drawing tasks with great efficiency.  To make it possible to utilize
+such facilities in the most efficient fashion, LVGL v9 and onwards implements a
+"Drawing Pipeline", like an assembly line, where decisions are made as to which
+drawing tasks drawing tasks (:ref:`Draw Tasks`) are given to which "logic entities"
+(:ref:`Draw Units`) in order to be carried out.
+
+This Pipeline is designed so that it is both flexible and extendable.  As a
+programmer, you can hook into it in order to provide LVGL with guidance as to what
+Draw Units should receive what types of Draw Tasks, or replace LVGL's built-in
+software rendering logic to any degree you choose.
+
+
+.. _draw tasks:
+
+Draw Tasks
+----------
+
+A "Draw Task" (:cpp:type:`lv_draw_task_t`) is a package of information that is
+created at the beginning of the Drawing Pipeline when a request to draw is made.
+Functions such as :cpp:expr:`lv_draw_rect()` and :cpp:expr:`lv_draw_label()` create
+one or more Draw Tasks and pass them down the Drawing Pipeline.  Each Draw Task
+carries all the information required to:
+
+- compute which :ref:`Draw Unit <draw units>` should receive this task, and
+- give the Draw Unit all the details required accomplish the drawing task.
+
+A Draw Task carries the following information:
+
+- type:  defines the drawing algorithm involved (e.g. line, fill, border, image,
+  label, arc, triangle, etc.)
+- area:  defines the rectangle involved where drawing will occur
+- transformation matrix:  if LV_DRAW_TRANSFORM_USE_MATRIX is configured to '1'
```

Comment:
```suggestion
- transformation matrix:  if ``LV_DRAW_TRANSFORM_USE_MATRIX`` is configured to '1'
```

Or maybe  `:cpp:macro:` can be used here too.

Reply:
> Or maybe `:cpp:macro:` can be used here too.

Good catch!  For whatever reason `:cpp:macro:` isn't implemented:

![image](https://github.com/user-attachments/assets/137089e3-d81e-4de2-8cd8-9e669281052f)

but `:c:macro:` is:

![image](https://github.com/user-attachments/assets/c19584e3-51b7-4deb-9daf-9d5cc9458b74)

Done and submitted.

Reply:
That's part of the issues with breathe. it has all of the directives and roles all kinds of screwed up.